Tests/PackageLoadingTests/ModuleMapGenerationTests.swift

```diff
@@ -128,7 +131,47 @@ class ModuleMapGeneration: XCTestCase {
         checkExpected()
     }
 
+    func testDontRewriteUnlessChanged() throws {
+        // Test that modulemap is not re-written to disk when its contents aren't updated.
+        // We use real file system for this test case to read the modification timestamp.
+        mktmpdir { dir in
+            // Create dummy module.
+            let module = try ClangModule(name: "Foo", isTest: false, sources: Sources(paths: [], root: dir))
+            var generator = ModuleMapGenerator(for: module)
+
+            // Create files.
+            try localFileSystem.createDirectory(dir.appending(components: "include"))
+            try localFileSystem.writeFileContents(dir.appending(components: "include", "Foo.h"), bytes: ByteString())
+
+            // Generate modulemap.
+            try generator.generateModuleMap(inDir: dir)
+
+            let moduleMap = dir.appending(components: "module.modulemap")
+            let contents = try localFileSystem.readFileContents(moduleMap)
+
+            XCTAssert(localFileSystem.isFile(moduleMap))
+            // Store modified time.
+            let modifiedTime = try getModifiedTime(file: moduleMap)
+            // Need to sleep for 1 second because time is precisely stored in seconds.
+            sleep(1)
```

Comment:
Even then sleep has to be there right? for eg: https://github.com/apple/swift-package-manager/blob/master/Tests/FunctionalTests/MiscellaneousTests.swift#L249